packages-next/cloudinary/src/Implementation.ts

```diff
@@ -93,31 +145,40 @@ class CloudinaryImage extends Implementation {
   // Called on `User.avatar` for example
   gqlOutputFieldResolvers() {
     return {
-      [this.path]: item => {
-        let itemValues = item[this.path];
-        if (!itemValues) {
+      [this.path]: (item: Record<P, undefined | null | string | File>) => {
+        let itemValues: undefined | null | string | File = item[this.path];
+        if (itemValues === null || itemValues === undefined) {
           return null;
         }
         if (this.adapter.listAdapter.parentAdapter.provider === 'sqlite') {
           // we store document data as a string on sqlite because Prisma doesn't support Json on sqlite
           // https://github.com/prisma/prisma/issues/3786
           try {
-            itemValues = JSON.parse(itemValues);
+            itemValues = JSON.parse(itemValues as string) as File;
           } catch (err) {}
         }
-
+        const _itemValues = itemValues as File;
         return {
-          publicUrl: this.fileAdapter.publicUrl(itemValues),
-          publicUrlTransformed: ({ transformation }) =>
-            this.fileAdapter.publicUrlTransformed(itemValues, transformation),
-          ...itemValues,
+          publicUrl: this.fileAdapter.publicUrl(_itemValues),
+          publicUrlTransformed: ({ transformation }: { transformation: CloudinaryImageFormat }) =>
+            this.fileAdapter.publicUrlTransformed(_itemValues, transformation),
+          ..._itemValues,
         };
       },
     };
   }
 
-  async resolveInput({ resolvedData, existingItem }) {
-    const previousData = existingItem && existingItem[this.path];
+  async resolveInput({
+    resolvedData,
+    existingItem,
+  }: {
+    resolvedData: Record<
+      P,
+      { createReadStream: () => fs.ReadStream; filename: string; mimetype: any; encoding: any }
```

Comment:
This type should be `FileUpload` from `'graphql-upload'`

Reply:
Ah, yes, thank you!